### ISSUE TITLE
Memory: add optional expiration date to entries

### ISSUE THREAD

**Issue by clement-cunin** [clement-cunin/JeedomAlfred#52]
## Context

Memory entries in `alfred_memory` are currently permanent. Some entries (daily journals, digests) should expire automatically after a configurable number of days.

### DB migration
- Add nullable column `expires_at DATETIME DEFAULT NULL` to `alfred_memory`

### `alfredMemory` class
- `save()` gains optional `$expiresAt` parameter (`?string`)
- `updateByLabel()` and `adminUpdate()` can set or clear expiration
- `loadForUser()` filters out expired entries: `WHERE expires_at IS NULL OR expires_at > NOW()`
- `cronDaily()` runs a cleanup query to delete expired rows and keep the table small

### Memory tools exposed to LLM
- `alfred_memory_save`: add optional `expires_in_days` integer field
- `alfred_memory_update`: add optional `expires_in_days` field (pass `0` to clear expiration)

### Prompt injection
Show expiration next to the label in the memory block:
`### journal-clement-2026-06-04 *(expires 2026-06-14)*`

## Notes
- Expiration is checked at read time (`loadForUser`), not by a separate background job
- Expired rows are physically deleted during `cronDaily()` to keep the table lean

**Comment by clement-cunin** [comment]
## ⚠️ AutoClaude agent crash

The agent exited with status **error** and no PR was created.

```
[Claude] I'll start by exploring the codebase to understand the existing structure before implementing the changes.
[Tool:Agent] {"description":"Explore alfredMemory codebase structure","subagent_type":"Explore","prompt":"Search for all files related to alfredMemory and memory tools in the JeedomAlfred project at /workspace/JeedomAlfred. Find:\n1. The alfredMemory class file\n2. Any DB migration files\n3. Memory-related tool definitions (LLM tools)\n4. The cronDaily method\n5. The loadForUser method\n6. save(), updateByLabe
[Tool:Bash] {"command":"find /workspace/JeedomAlfred -type f -name \"*.php\" -o -name \"*.js\" -o -name \"*.json\" | head -20"}
[Tool:Glob] {"pattern":"**/*[Mm]emory*","path":"/workspace/JeedomAlfred"}
[Tool:Grep] {"pattern":"alfredMemory|class Memory","path":"/workspace/JeedomAlfred","type":"php","output_mode":"files_with_matches"}
[Tool:Grep] {"pattern":"cronDaily|loadForUser|save\\(\\)|updateByLabel|adminUpdate","path":"/workspace/JeedomAlfred","type":"php","output_mode":"files_with_matches"}
[Tool:Glob] {"pattern":"**/*migration*","path":"/workspace/JeedomAlfred"}
[Tool:Glob] {"pattern":"**/*migrate*","path":"/workspace/JeedomAlfred"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredMemory.class.php"}
[Tool:Bash] {"command":"find /workspace/JeedomAlfred -type f \\( -name \"*.sql\" -o -name \"*install*\" -o -name \"*update*\" \\) | grep -v node_modules"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/plugin_info/install.php"}
[Tool:Grep] {"pattern":"memory.*tool|tool.*memory|remembrance|saveMemory|updateMemory|forgetMemory|memoryManagement|listMemory","path":"/workspace/JeedomAlfred","type":"php","output_mode":"files_with_matches"}
[Tool:Glob] {"pattern":"**/*Migration*","path":"/workspace/JeedomAlfred"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredAgent.class.php","limit":300}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredMigration.class.php"}
[Tool:Bash] {"command":"grep -n \"cronDaily\\|memory\" /workspace/JeedomAlfred/core/class/alfredAgent.class.php | head -40"}
[Tool:Bash] {"command":"grep -n \"cronDaily\" /workspace/JeedomAlfred/core/class/*.php"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredAgent.class.php","offset":330,"limit":150}
[Tool:Bash] {"command":"grep -n \"cronDaily\\|cron_daily\" /workspace/JeedomAlfred/core/class/alfred.class.php"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredAgent.class.php","offset":850,"limit":100}
[Tool:Bash] {"command":"grep -rn \"cronDaily\\|cron_daily\" /workspace/JeedomAlfred/core/class/"}
[Tool:Bash] {"command":"grep -rn \"function cron\" /workspace/JeedomAlfred/core/class/"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfred.class.php","limit":200}
[Tool:Bash] {"command":"find /workspace/JeedomAlfred -type f -name \"*.php\" | xargs grep -l \"cronDaily\\|cron_daily\" 2>/dev/null"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/plugin_info/configuration.php"}
[Tool:Bash] {"command":"grep -n \"listMemories\\|createMemory\\|updateMemory\\|deleteMemory\\|adminUpdate\" /workspace/JeedomAlfred/core/ajax/alfred.ajax.php | head -30"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/ajax/alfred.ajax.php","limit":150}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/ajax/alfred.ajax.php","offset":150,"limit":60}
[Tool:Bash] {"command":"grep -rn \"function cron\\|cronDaily\\|public static function cron\" /workspace/JeedomAlfred/core/class/ 2>/dev/null"}
[Tool:Bash] {"command":"ls -la /workspace/JeedomAlfred/core/sql/ 2>/dev/null"}
[Claude] Now let me read the key files to understand the exact implementation:
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredMemory.class.php"}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredMigration.class.php","limit":100}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredMigration.class.php","offset":100}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredAgent.class.php","offset":320,"limit":120}
[Claude] Now let me read the system prompt building and memory tool handling parts of alfredAgent:
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfredAgent.class.php","offset":790,"limit":120}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfred.class.php","offset":80,"limit":40}
[Claude] Now let me look at the full alfred.class.php to understand cron methods and also the configuration UI:
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/class/alfred.class.php","limit":90}
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/plugin_info/configuration.php","offset":220,"limit":100}
[Claude] Now let me read the JavaScript memory management section:
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/plugin_info/configuration.php","offset":940,"limit":230}
[Claude] Let me read the AJAX handler for memory operations:
[Tool:Read] {"file_path":"/workspace/JeedomAlfred/core/ajax/alfred.ajax.php","offset":145,"limit":60}
[Claude] Now I have a complete picture. Let me implement all the changes systematically.
[Tool:Edit] {"replace_all":false,"file_path":"/workspace/JeedomAlfred/core/class/alfredMigration.class.php","old_string":"    const MIGRATIONS = [\n        1 => 'migration_001_baseline',\n        2 => 'migration_002_async_tasks',\n        3 => 'migration_003_push_notifications',\n    ];","new_string":"    const MIGRATIONS = [\n        1 => 'migration_001_baseline',\n        2 => 'migration_002_async_tasks',\n
[Tool:Edit] {"replace_all":false,"file_path":"/workspace/JeedomAlfred/core/class/alfredMigration.class.php","old_string":"    private static function migration_003_push_notifications_down(): string\n    {\n        return implode(\";\\n\", [\n            'DROP TABLE IF EXISTS `alfred_push_notification`',\n            'DROP TABLE IF EXISTS `alfred_push_subscription`',\n        ]);\n    }\n}","new_string":"    p
[Claude] You've hit your session limit · resets 10:20pm (UTC)
[Usage] input=11 output=7324 cache_read=264128 cache_write=34161
```
